docs/contributor/03-11-hap-rules.md

```diff
@@ -0,0 +1,207 @@
+
+## Rule Attributes
+
+Triggered rule entry applies that labels to the SecretBinding search query.
+Rule entry attributes and its plan alter the search query by adding new or modifing existing search labels. Existence of an attribute in the rule means its inclusion in the search labels in addition to labels associated with rule plan.
+
+### PR Attribute
+
+It is possible to extend the rule entry with specified platform region by adding  a `PR` rule attribute. The value of PR attributes is compared to region of a subaccount of SKR and, if triggered, applies `hyperscalerType: <HYPERSCALER_NAME>[_<PLATFORM_REGION>]` label.
+
+Below configuration means that if a gcp cluster is provisioned in cf-sa30 subaccount region then KEB would search for secret bindings with `hyperscalerType: gcp_cf-sa30` label. 
+
+```
+hap: 
+  rule: 
+    - gcp(PR=cf-sa30)
+
+SecretBinding pools
+- hyperscalerType: gcp_cf-sa30, 
+```
+
+### Cluster Region Attribute
+
+A region that an SKR is provisioned in can be matched with an attribute named `CR`. Below configuration specifies azure plan as one for which to use cluster region based search. In this case, cluster provisioned in with the azure plan would require a secret binding with `hyperscalerType: azure_<CLUSTER_REGION>` label. 
+
+```
+hap: 
+  rule: 
+    - gcp(CR=us-central1)
+
+SecretBinding pools
+- hyperscalerType: gcp_us-central1, 
+```
+
+### Shared & EU Access attribute
+
+The `shared` and `euAccess` attributes does not correspond to any physical SKR property. It is a logical attribute that is used only to add its label. If the rule entry contains `shared` or `euAccess` attributes then when triggered a `shared: true` or `euAccess: true` labels are added to the search labels depending on which one of those two attributes is listed in the rule entry. Shared label on a SecretBindings marked it as assignable to more than one SKR and euAccess is a non-functional label not reused by KEB (more on that can be found in p[HAP Pool document](03-10-hyperscaler-account-pool.md)). Below configuration specifies that all gcp clusters will use the same pool of shared secret bindings marked with labels `hyperscalerType: gcp`, `shared: true` and azure clusters in the region cf-ch20 will use the pool of secret bindings marked with labels `hyperscalerType: azure`, `euAccess: true`. 
+
+```
+hap: 
+  rule: 
+    - gcp(S)
+    - azure(PR=cf-ch20, E)
+
+SecretBinding pools:
+- hyperscalerType: gcp; shared: true
+- hyperscalerType: azure_cf-ch20, euAccess: true
+```
+
+### Attribute with "*"
+
+Attributes values can be replaced with `*` which means that all SKR values of that attribute are matched and reused in unchanged form when applying search labels modifications. For example, the following configuration makes KEB search SecretBindings with `hyperscalerType: gcp_<PR>`, where `PR` corresponds to an SKR value, for all `gcp` clusters:
+
+```
+hap: 
+  rule: 
+  - gcp(PR=*)
```

Comment:
Input) gcp PR=x CR=y

Rule 1):
gcp(PR=*) - output: gcp_cf-x
gcp             - output: gcp

Validation ok? - fail. Say that input rules are not unique

Rule 2):
gcp(PR=*, CR=*) - output: gcp_cf-x_westeu
gcp.                      - output: gcp

Validation ok? - fail. Say that input rules are not unique

---
Rule)
gcp(PR=*)
gcp(CR=westeu)
gcp(PR=*, CR=westeu)

When validation, remove all PR=* and CR=* to check uniqness